src/passes/hash-stringify-walker.cpp

```diff
@@ -147,6 +148,45 @@ std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::dedupe(
   return result;
 }
 
+std::vector<SuffixTree::RepeatedSubstring> StringifyProcessor::removeOverlaps(
+  const std::vector<SuffixTree::RepeatedSubstring>& substrings) {
+  std::vector<Interval> intervals;
+  std::unordered_map<Interval, unsigned> intervalMap;
+
+  // Construct intervals
+  for (Index i = 0; i < substrings.size(); i++) {
+    auto substring = substrings[i];
+    for (auto startIdx : substring.StartIndices) {
+      auto interval =
+        Interval(startIdx,
+                 startIdx + substring.Length - 1,
+                 substring.Length * substring.StartIndices.size());
+      intervals.push_back(interval);
+      intervalMap[std::move(interval)] = i;
+    }
+  }
+
+  // Get the overlapping intervals
+  std::set<Interval> overlaps = IntervalProcessor::getOverlaps(intervals);
+  std::set<unsigned> doNotInclude;
+  for (auto& interval : overlaps) {
+    doNotInclude.insert(intervalMap[interval]);
+  }
```

Comment:
We could simplify the code here and get away without any map or set lookups if `IntervalProcessor` returned a sequence of kept indices in its input vector rather than a set of removed intervals. With a sequence of kept indices, we could directly construct the list of kept substrings.

Reply:
great idea, thanks!